src/wasm.h

```diff
@@ -1172,7 +1172,39 @@ struct BinaryLocations {
   struct Span {
     BinaryLocation start, end;
   };
+
+  // Track the range of addresses an expressions appears at. This is the
+  // contiguous range that all instructions have - control flow instructions
+  // have additional opcodes later (like an end for a block or loop), see
+  // just after this.
   std::unordered_map<Expression*, Span> expressions;
+
+  // Track the extra offsets that some instructions, in particular control
+  // flow, have, like 'end' for loop and block. We keep these in a separate map
+  // because they are rare and we optimize for the storage space for the common
+  // type of instruction which just needs a Span.
+  // We implement this as a simple struct with two elements (as two extra
+  // elements is the maximum currently needed; due to 'catch' and 'end' for
+  // try-catch). The second value may be 0, indicating it is not used.
+  struct ExtraLocations : public std::array<BinaryLocation, 2> {
```

Comment:
"extra" naming everywhere in this PR isn't very descriptive. Really these extra locations and opcodes are related to the block structure (or control structure) of the IR. So something like "block/control structure locations" or "block delimiter locations" or some kind of similar name would make more sense.

Reply:
"Extra" here is meant to mean "in addition to the main part of the instruction", some extra locations for it. But I agree it's not super-clear.

"Structure locations", "delimiter locations" worry me in that they don't indicate it's the *extra* ones aside from the initial main part (which is also part of the structure).

Reply:
the "delimiter" bit was meant to get that while still including "catch" and "else". I think "delimiter locations" has that implication for me (i.e. it's not the main part). We could just call it something like StructureEndLocations or BlockEndLocations and say in the comments that "end" means the end of any kind of block-structured construct. For me it's more important for the name to emphasize something about what this thing actually means/is rather than emphasize that it's extra, so if we have to have one or the other, i'd prefer that.

Reply:
I see what you mean. Delimiters sounds good then (as it captures "else" and "catch" as well, which are not at the end)?